domain/src/test/java/org/oppia/android/domain/auth/AuthenticationModuleTest.kt

```diff
@@ -29,51 +28,37 @@ import javax.inject.Singleton
 @Suppress("FunctionName")
 @RunWith(AndroidJUnit4::class)
 @LooperMode(LooperMode.Mode.PAUSED)
-@Config(manifest = Config.NONE)
+@Config(application = AuthenticationModuleTest.TestApplication::class)
 class AuthenticationModuleTest {
 
   @Inject
-  lateinit var wrapper: AuthenticationWrapper
+  lateinit var firebaseAuthWrapper: FirebaseAuthWrapper
 
   @Before
   fun setUp() {
     setUpTestApplicationComponent()
   }
 
   @Test
-  fun testModule_injectsInstanceOfAuthenticationWrapper() {
-    assertThat(wrapper).isInstanceOf(AuthenticationController::class.java)
+  fun testModule_injectsProductionImplementationOfFirebaseAuthWrapper() {
+    assertThat(firebaseAuthWrapper).isInstanceOf(FirebaseAuthWrapperImpl::class.java)
   }
 
   private fun setUpTestApplicationComponent() {
-    DaggerAuthenticationModuleTest_TestApplicationComponent.builder()
-      .setApplication(ApplicationProvider.getApplicationContext())
-      .build()
-      .inject(this)
-  }
-
-  @Module
-  class TestModule {
-    @Provides
-    @Singleton
-    fun provideContext(): Context {
-      return ApplicationProvider.getApplicationContext()
-    }
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
   }
 
+  // TODO(#89): Move this to a common test application component.
   @Module
-  class AuthenticationModule {
-    @Provides
-    @Singleton
-    fun provideAuthenticationController():
-      AuthenticationWrapper = AuthenticationController(mock(FirebaseAuth::class.java))
+  interface TestModule {
+    @Binds
+    fun provideContext(application: Application): Context
   }
 
-  // TODO(#89): Move this to a common test application component.
   @Singleton
   @Component(
     modules = [
-      TestModule::class, TestDispatcherModule::class, AuthenticationModule::class,
+      TestModule::class, TestDispatcherModule::class, TestAuthenticationModule::class,
```

Comment:
Unable to get this test to pass since it requires `AuthenticationModule::class`, which still requires the real firebase app to be initialized. I am experiencing difficulty providing a fake while still Binding the ProdImpl of the wrapper, to be tested.

Reply:
Why can't we bind the production implementations here for this test? Not clear to me what the specific issue is.